client/verta/verta/pipeline/_pipeline_graph.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+
+from typing import Any, Dict, List
+
+from verta._internal_utils._utils import Configuration, Connection
+from ._pipeline_step import PipelineStep
+
+
+class PipelineGraph:
+    """
+    A collection of PipelineSteps to be run as a single inference pipeline.
+
+    Parameters
+    ----------
+    steps : list of :class:`~verta.pipeline.PipelineStep`
+        List of all possible steps of the pipeline. Ordering of steps in the pipeline
+        itself is determined by the predecessors provided to each step, thus ordering
+        of this list is irrelevant.
+
+    Attributes
+    ----------
+        steps: list of :class:`~verta.deployment.PipelineStep`
+            List of PipelineSteps comprising all possible steps in the PiplineGraph.
+    """
+
+    def __init__(self, steps: List[PipelineStep]):
+        self._steps = steps
```

Comment:
We should probably use the validation logic we have in `set_steps()`

```suggestion
        self._steps = self.set_steps(steps)
```